staging/src/k8s.io/kubelet/pkg/apis/podresources/v1/api.proto

```diff
@@ -18,6 +18,15 @@ option (gogoproto.goproto_unrecognized_all) = false;
 // node resources consumed by pods and containers on the node
 service PodResourcesLister {
     rpc List(ListPodResourcesRequest) returns (ListPodResourcesResponse) {}
+    rpc GetAllocatableResources(AllocatableResourcesRequest) returns (AllocatableResourcesResponse) {}
+}
+
+message AllocatableResourcesRequest {}
+
+// AllocatableResourcesResponses contains informations about all the devices known by the kubelet
+message AllocatableResourcesResponse {
+    repeated ContainerDevices devices = 1;
+    repeated int64 cpu_ids = 2;
```

Comment:
Why not just have this return a `repeated ContainerResources` instead of duplicating the fields of this in here?

Reply:
I need to doublecheck the revision history, but I think we can change as you suggested, which seems nicer indeed.

Reply:
`AllocatableResourcesResponse`  should return cumulative data of all the devices and cpus known to kubelet i.e. data corresponding to not only resources allocated to containers but also the ones that haven't been allocated. Therefore, we should keep it the way we have it now instead of using `repeated ContainerResources` because changing to `repeated ContainerResources` would mean we would only account for resources that have been allocated to containers.

Reply:
Thanks @swatisehgal! It boils down to how resources are accounted indeed and how we report to clients; yes, there are similarities, but reusing the `ContainerResources`, albeit doable (the fields are indeed similar) will make harder for the clients to consume the data. Therefore I'm keeping the current API.